lib/aws/session_store/dynamo_db/locking/base.rb

```diff
@@ -13,8 +13,14 @@ def set_session_data(env, sid, session, options = {})
       return false if session.empty?
       packed_session = pack_data(session)
       handle_error(env) do
-        save_opts = update_opts(env, sid, packed_session, options)
-        @config.dynamo_db_client.update_item(save_opts)
+        if env['dynamo_db.new_session']
```

Comment:
If we're removing pessimistic locking, does base/null locking as classes even make sense? We can possibly refactor away anything related to locking.

Reply:
I don't think I need that feature, but others might. I cannot make the decision, but I don't have much time to implement and test for the feature. Please rename files to a suitable names if we could remove the feature.

Reply:
I think it's fine to remove pessimistic locking. I am saying that `locking/base` and `locking/null` are no longer relevant (because locking is being removed entirely) and we can probably delete these classes and put the logic into rack_middleware or some other new class.